packages/flutter_test/lib/src/binding.dart

```diff
@@ -1361,7 +1361,7 @@ class AutomatedTestWidgetsFlutterBinding extends TestWidgetsFlutterBinding {
     final Zone realAsyncZone = Zone.current.fork(
       specification: ZoneSpecification(
         scheduleMicrotask: (Zone self, ZoneDelegate parent, Zone zone, void Function() f) {
-          Zone.root.scheduleMicrotask(f);
+          _rootDelegate.scheduleMicrotask(zone, f);
```

Comment:
This could have just done `Zone.root.scheduleMicrotask(zone.bindCallbackGuarded(f))`, but that's not nevessarily the same as what `rootDelegate.scheduleMicrotask(zone, f)` does.
The `bindCallbackGuarded` does an extra `Zone.registerCalback`, and an extra `Zone.run` on top of what the root zone's `scheduleMicrotask` would already do.

Using the root zone's delegate is ensures that it does exactly the same as if `zone` had been a direct child of the root zone, just bypassing the intermediate zones in order to get "real" async operations.